pkg/server/api/agent/v1/service.go

```diff
@@ -39,7 +44,49 @@ func (s *Service) ListAgents(ctx context.Context, req *agent.ListAgentsRequest)
 }
 
 func (s *Service) GetAgent(ctx context.Context, req *agent.GetAgentRequest) (*types.Agent, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	reqID, err := api.IDFromProto(req.Id)
+	if err != nil {
+		log.WithError(err).Error("Failed to parse SPIFFE ID")
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	resp, err := s.ds.FetchAttestedNode(ctx, &datastore.FetchAttestedNodeRequest{
+		SpiffeId: reqID.String(),
+	})
+	if err != nil {
+		log.Errorf("Failed to fetch node: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to fetch node: %v", err)
+	}
+
+	if (resp.Node) == nil {
+		log.Error("Agent not found")
+		return nil, status.Error(codes.NotFound, "agent not found")
+	}
+
+	selectors, err := s.getSelectorsFromAttestedNode(ctx, resp.Node)
+	if err != nil {
+		log.Errorf("Failed to get selectors from attested node: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get selectors from attested node: %v", err)
+	}
+
+	spiffeID, err := spiffeid.FromString(resp.Node.SpiffeId)
+	if err != nil {
+		return nil, fmt.Errorf("node has malformed SPIFFE ID: %v", err)
+	}
+
+	agent := &types.Agent{
+		Id:                   api.ProtoFromID(spiffeID),
+		AttestationType:      resp.Node.AttestationDataType,
+		X509SvidSerialNumber: resp.Node.CertSerialNumber,
+		X509SvidExpiresAt:    resp.Node.CertNotAfter,
+		Selectors:            selectors,
+		Banned:               resp.Node.CertSerialNumber == "" && resp.Node.NewCertSerialNumber == "",
```

Comment:
I'm not sure if this is the right logic to determine a banned agent?

Reply:
Maybe I'm confused about the role of `NewCertSerialNumber`. I'd love the feedback from @azdagron here. This also impacts  https://github.com/spiffe/spire/pull/1667.

Reply:
I think `CertSerialNumber` is good enough to know that an agent is a banned. However, I still think BanAgent should set both to empty. This prevents a renewed-but-not-yet-reconnected agent from coming back in after a call to ban.